lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1613,8 +1614,10 @@ def plot_surface(self, X, Y, Z, *args, norm=None, vmin=None,
            fcolors is None:
             polys = np.stack(
                 [cbook._array_patch_perimeters(a, rstride, cstride)
-                 for a in (X, Y, Z)],
+                 for a in (X, Y, Z, mask)],
```

Comment:
Is it really necessary to stack `mask` into `polys`, when it's going to be split apart anyway on the next lines?

Reply:
It isn't really necessary. I thought it was neater because otherwise we need another `cbook._array_patch_perimeters` line for the masks. I don't have a strong opinion, so happy to change if that is the preferred option?

Reply:
We already know every thing is the right size (I think?), so splitting them up would save some broadcast checks and copies.

Reply:
I've moved the mask outside of the `np.stack`